cirq-core/cirq/optimizers/transformer_primitives_test.py

```diff
@@ -198,3 +200,126 @@ def test_map_moments_drop_empty_moments():
     c = cirq.Circuit(cirq.Moment(op), cirq.Moment(), cirq.Moment(op))
     c_mapped = cirq.map_moments(c, lambda m, i: [] if len(m) == 0 else [m])
     cirq.testing.assert_same_circuits(c_mapped, cirq.Circuit(c[0], c[0]))
+
```

Comment:
Can we add a unit test to verify runtime on some larg-ish circuits, just to make sure as work with `merge_***` we don't accidentally turn something into N^2 ?

Reply:
Added tests to verify the number of incoming calls to `merge_func`.